code/game/machinery/vendors/vendor_crits.dm

```diff
@@ -0,0 +1,115 @@
+/**
+ * Framework for custom vendor crits.
+ */
+
+/datum/vendor_crit
+	/// If it'll deal damage or not
+	var/harmless = FALSE
+	/// If we should be thrown against the mob or not.
+	var/fall_towards_mob = TRUE
+
+/**
+ * Return whether or not the crit selected is valid.
+ */
+/datum/vendor_crit/proc/is_valid(obj/machinery/economy/vending/machine, mob/living/carbon/victim)
+	return TRUE
+
+/***
+ * Perform the tip crit effect on a victim.
+ * Arguments:
+ * * machine - The machine that was tipped over
+ * * user - The unfortunate victim upon whom it was tipped over
+ * Returns: The "crit rebate", or the amount of damage to subtract from the original amount of damage dealt, to soften the blow.
+ */
+/datum/vendor_crit/proc/tip_crit_effect(obj/machinery/economy/vending/machine, mob/living/carbon/victim)
+	return 0
+
+/datum/vendor_crit/shatter
+
+/datum/vendor_crit/shatter/tip_crit_effect(obj/machinery/economy/vending/machine, mob/living/carbon/victim)
+	victim.bleed(150)
+	var/obj/item/organ/external/leg/right = victim.get_organ(BODY_ZONE_R_LEG)
+	var/obj/item/organ/external/leg/left = victim.get_organ(BODY_ZONE_L_LEG)
+	if(istype(left))
+		left.receive_damage(200)
+	if(istype(right))
+		right.receive_damage(200)
+
+	if(left || right)
+		victim.visible_message(
+			"<span class='danger'>[victim]'s legs shatter with a sickening crunch!</span>",
+			"<span class='userdanger'>Your legs shatter with a sickening crunch!</span>",
+			"<span class='danger'>You hear a sickening crunch!</span>"
+		)
+
+	return machine.squish_damage - 10
```

Comment:
Why use `machine.squish_damage` here if you're doing substractions instead of taking a percentage of it or so.
If `squish_damage` gets changed to 10 then some of these crit effects will heal the user.